### DISCUSSION TITLE
Creating an package initializer

### DISCUSSION THREAD

**Discussion by johnhooks** [erikyo/typescript-wp-block#7] [Ideas]
I think package would be most powerful as an initializer. Where it adds in all the correct information for the newly created project, rather than leave it to the user to fill in.

Here is an example of one I have created before:

https://github.com/johnhooks/create-pkg

The advantage of this method, over how Gutenberg does it, the files aren't templates, so all the regular IDE features still work.

Let me know if you think this would be a good approach, if so, I'll start working on it.

**Comment by johnhooks** [comment]
It uses string replacement to replace all the "template" variables that are in the format of `~ALL_CAPS~`.

**Comment by erikyo** [comment]
yup I see! Is a really good way to scaffold a template but I was wondering if is possibile to make an npm package instead of nesting the block in a sub folder

**Comment by johnhooks** [comment]
Actually, I am considering rewriting `@wordpress/create-block` and use this method, and then we can have many templates in their own packages.

**Comment by johnhooks** [comment]
Also putting it all in a mono repo.

**Comment by erikyo** [comment]
I am moving this to the discussions because it is an interesting point that I would like to develop, although as I said I would like at the same time to keep the repo as easy to get into as possible

**Comment by johnhooks** [comment]
I totally understand keeping this repo as clean as possible, as soon as templates or monorepos are used it may become easier to maintain, but way more complex for anyone to help with it. I have the npm and github organization `wp-blocks` registered. I'm considering creating a monorepo and publish all the npm packages under the `@wp-blocks/*` namespace. I believe I invited you to be a admin of that github organization. Think about it, it might be a good place for the intense TypeScript "developer" version to live, and the more accessible version could live here. Just thoughts.

**Comment by erikyo** [comment]
it's a cool project, I'll join right away of course! and besides this way it makes sense, an organisation is definitely the best way to organise multiple repos and have them in the same place.

**Comment by johnhooks** [comment]
I made you an owner of that organization too. We're going to do this together!

**Comment by johnhooks** [comment]
I'm working on the initial concept of what main monorepo will look like. I'll try to get it up soon so you can see what that looks like.
